optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/header/HeaderView.java

```diff
@@ -16,14 +16,17 @@
 
 package org.optaplanner.openshift.employeerostering.gwtui.client.header;
 
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
 import elemental2.dom.HTMLAnchorElement;
+import elemental2.dom.HTMLDivElement;
 import org.jboss.errai.ui.client.local.api.elemental2.IsElement;
 import org.jboss.errai.ui.shared.api.annotations.DataField;
 import org.jboss.errai.ui.shared.api.annotations.Templated;
 
-import javax.inject.Inject;
-
 @Templated
+@ApplicationScoped
```

Comment:
Why did you have to make it `@ApplicationScoped`?

Reply:
It need to be a singleton since the roster page uses it to "add on" their toolbars to the "headers" (which allow them to be correctly positioned and sticky w/o setting their top values) (won't work for only vertically sticky elements, such as date times)